src/components/OrdersContainer/Orders/OrderForm/Fields/Field/Field.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+    NumericType,
+    isSignedIntegerType,
+    isUnsignedIntegerType,
+} from "adapters/GolangTypes";
+import { NumericValue } from "adapters/Order";
+type Props = {
+    name: string;
+    field: FieldData;
+    onChange: (newValue: boolean | string | number, isValid: boolean) => void;
+    changeEnabled: (isEnabled: boolean) => void;
+};
+
+export const Field = ({ name, field, onChange, changeEnabled }: Props) => {
+    function handleTextInputChange(value: string) {
+        //FIXME: mergear tipos Value y ValueType o algo así
+        const isValid = isNumberValid(
+            value,
+            (field.valueDescription as NumericValue).value
+        );
+
+        onChange(Number.parseFloat(value), isValid);
+    }
+
+    return (
+        <div
+            className={`${styles.fieldWrapper} ${
+                !field.isEnabled ? styles.disabled : ""
+            }`}
+        >
+            <div className={styles.name}>{name}</div>
+            {field.valueDescription.kind == "numeric" ? (
+                <TextInput
+                    placeholder={`${field.valueDescription.value}...`}
+                    defaultValue={!field.isValid ? "" : (field.value as number)}
+                    isRequired={field.isEnabled}
+                    isEnabled={field.isEnabled}
+                    isValid={field.isValid}
+                    onChange={handleTextInputChange}
+                ></TextInput>
+            ) : field.valueDescription.kind == "boolean" ? (
+                <CheckBox
+                    isRequired={field.isEnabled}
+                    disabled={!field.isEnabled}
+                    onChange={(value: boolean) => {
+                        onChange(value, true);
+                    }}
+                />
+            ) : (
+                <Dropdown
+                    options={field.valueDescription.value}
+                    defaultValue={field.value as string}
+                    onChange={(newValue) => {
+                        onChange(newValue, true);
+                    }}
+                />
+            )}
+
+            <CheckBox
+                color="orange"
+                isRequired={true}
+                onChange={changeEnabled}
+                initialValue={field.isEnabled}
+            />
+        </div>
+    );
+};
+
+function isNumberValid(valueStr: string, numberType: NumericType): boolean {
+    return (
+        checkNumberString(valueStr, numberType) &&
+        ((isUnsignedIntegerType(numberType) &&
+            checkUnsignedIntegerOverflow(
+                Number.parseInt(valueStr),
+                getBits(numberType)
+            )) ||
+            (isSignedIntegerType(numberType) &&
+                checkSignedIntegerOverflow(
+                    Number.parseInt(valueStr),
+                    getBits(numberType)
+                )) ||
+            checkFloatOverflow(Number.parseFloat(valueStr)))
+    );
+}
+
```

Comment:
is it appropriate to have all these functions in this tsx?

Reply:
I will make a separate file for field valiadation